### ISSUE TITLE
Slow first start

### ISSUE THREAD

**Issue by sylr** [cilium/cilium#24832]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

In an **AWS EKS cluster** where we run **custom AMI** where we have preloaded with the cluster's deamonset container images (such as cilium's) we observe that cilium is slow to start the first time it is executed on a new node.

This slowness only occurs the first time cilium is executed on a node, a cilium restart is blazing fast.

We don't have any resource limits (cpu, memory) set on the cilium pod.

The sysdump is way too big to be uploaded here, if anyone is interested to investigate I would be happy to give you anything needed, _you can contact me on cilium's Slack for more synchronous interactions_.

### Cilium Version

v1.13.1

### Kernel Version

5.15.93

### Kubernetes Version

1.25.6

### Relevant log output

```shell
2023-04-12T08:43:13.709182953Z level=info msg="     _ _ _" subsys=daemon
2023-04-12T08:43:13.709184796Z level=info msg=" ___|_| |_|_ _ _____" subsys=daemon
2023-04-12T08:43:13.709186293Z level=info msg="|  _| | | | | |     |" subsys=daemon
2023-04-12T08:43:13.709187801Z level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon
2023-04-12T08:43:13.709189396Z level=info msg="Cilium 1.13.1 a6be57eb 2023-03-15T19:39:01+01:00 go version go1.19.6 linux/arm64" subsys=daemon
2023-04-12T08:43:19.410007515Z level=info msg="cilium-envoy  version: 04413917ff99e4f6ab51d1c6eb424d4a055f4462/1.23.4/Distribution/RELEASE/BoringSSL" subsys=daemon
2023-04-12T08:43:28.020286413Z level=info msg="clang (10.0.0) and kernel (5.15.93) versions: OK!" subsys=linux-datapath
2023-04-12T08:43:32.767768005Z level=info msg="linking environment: OK!" subsys=linux-datapath
2023-04-12T08:43:33.033950211Z level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
2023-04-12T08:43:33.033976750Z level=info msg="Mounted cgroupv2 filesystem at /run/cilium/cgroupv2" subsys=cgroups
2023-04-12T08:43:33.033979972Z level=info msg="Parsing base label prefixes from default label list" subsys=labels-filter
2023-04-12T08:43:33.035870508Z level=info msg="Parsing additional label prefixes from user inputs: []" subsys=labels-filter
2023-04-12T08:43:33.035886706Z level=info msg="Final label prefixes to be used for identity evaluation:" subsys=labels-filter
2023-04-12T08:43:33.035890535Z level=info msg=" - reserved:.*" subsys=labels-filter
2023-04-12T08:43:33.035892793Z level=info msg=" - :io\\.kubernetes\\.pod\\.namespace" subsys=labels-filter
2023-04-12T08:43:33.035894697Z level=info msg=" - :io\\.cilium\\.k8s\\.namespace\\.labels" subsys=labels-filter
2023-04-12T08:43:33.035896586Z level=info msg=" - :app\\.kubernetes\\.io" subsys=labels-filter
2023-04-12T08:43:33.035898504Z level=info msg=" - !:io\\.kubernetes" subsys=labels-filter
2023-04-12T08:43:33.035900362Z level=info msg=" - !:kubernetes\\.io" subsys=labels-filter
2023-04-12T08:43:33.035902196Z level=info msg=" - !:.*beta\\.kubernetes\\.io" subsys=labels-filter
2023-04-12T08:43:33.035904131Z level=info msg=" - !:k8s\\.io" subsys=labels-filter
2023-04-12T08:43:33.035905989Z level=info msg=" - !:pod-template-generation" subsys=labels-filter
2023-04-12T08:43:33.035907923Z level=info msg=" - !:pod-template-hash" subsys=labels-filter
2023-04-12T08:43:33.035909746Z level=info msg=" - !:controller-revision-hash" subsys=labels-filter
2023-04-12T08:43:33.035911593Z level=info msg=" - !:annotation.*" subsys=labels-filter
2023-04-12T08:43:33.035913531Z level=info msg=" - !:etcd_node" subsys=labels-filter
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sylr** [comment]
We've upgraded to v1.14.1 and the slowness problem seems to have disappeared:

```
2023-08-29T09:26:30.048546718Z level=info msg="     _ _ _" subsys=daemon
2023-08-29T09:26:30.048549358Z level=info msg=" ___|_| |_|_ _ _____" subsys=daemon
2023-08-29T09:26:30.048576200Z level=info msg="|  _| | | | | |     |" subsys=daemon
2023-08-29T09:26:30.048579560Z level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon
2023-08-29T09:26:30.048581741Z level=info msg="Cilium 1.14.1 c191ef6f 2023-08-10T18:54:57+02:00 go version go1.20.7 linux/amd64" subsys=daemon
2023-08-29T09:26:30.056845280Z level=info msg="clang (10.0.0) and kernel (5.15.110) versions: OK!" subsys=linux-datapath
2023-08-29T09:26:30.059418550Z level=info msg="linking environment: OK!" subsys=linux-datapath
2023-08-29T09:26:30.206202386Z level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
2023-08-29T09:26:30.206413571Z level=info msg="Mounted cgroupv2 filesystem at /run/cilium/cgroupv2" subsys=cgroups
2023-08-29T09:26:30.206419182Z level=info msg="Parsing base label prefixes from default label list" subsys=labels-filter
2023-08-29T09:26:30.206537790Z level=info msg="Parsing additional label prefixes from user inputs: []" subsys=labels-filter
2023-08-29T09:26:30.206544711Z level=info msg="Final label prefixes to be used for identity evaluation:" subsys=labels-filter
2023-08-29T09:26:30.206548291Z level=info msg=" - reserved:.*" subsys=labels-filter
2023-08-29T09:26:30.206550291Z level=info msg=" - :io\\.kubernetes\\.pod\\.namespace" subsys=labels-filter
2023-08-29T09:26:30.206551921Z level=info msg=" - :io\\.cilium\\.k8s\\.namespace\\.labels" subsys=labels-filter
2023-08-29T09:26:30.206553511Z level=info msg=" - :app\\.kubernetes\\.io" subsys=labels-filter
2023-08-29T09:26:30.206555092Z level=info msg=" - !:io\\.kubernetes" subsys=labels-filter
2023-08-29T09:26:30.206556592Z level=info msg=" - !:kubernetes\\.io" subsys=labels-filter
2023-08-29T09:26:30.206559082Z level=info msg=" - !:.*beta\\.kubernetes\\.io" subsys=labels-filter
2023-08-29T09:26:30.206561272Z level=info msg=" - !:k8s\\.io" subsys=labels-filter
2023-08-29T09:26:30.206563452Z level=info msg=" - !:pod-template-generation" subsys=labels-filter
2023-08-29T09:26:30.206565672Z level=info msg=" - !:pod-template-hash" subsys=labels-filter
2023-08-29T09:26:30.206579313Z level=info msg=" - !:controller-revision-hash" subsys=labels-filter
2023-08-29T09:26:30.206586394Z level=info msg=" - !:annotation.*" subsys=labels-filter
2023-08-29T09:26:30.206589074Z level=info msg=" - !:etcd_node" subsys=labels-filter
```

Would someone know what could explain this ?

**Comment by learnitall** [comment]
I'm sure somewhere along the way some kind of improvement was made to target this issue, but it may have been an indirect consequence of another change. It might be helpful to search through the commit logs.

**Comment by sylr** [comment]
Hi @learnitall, thank you for engaging. It turns out this issue might not cilium related after all.

I explained what I found in another issue https://github.com/awslabs/amazon-eks-ami/issues/1416. In a few words, cilium agent startup time seems to be related to whether or not the container image is already "pre-loaded" in the node that is starting. And, to my surprise, the slowness seems to appear when cilium's image is pre-loaded in the new node.
